### ISSUE TITLE
[Support:] Bluetooth only - or USB HID ?

### ISSUE THREAD

**Issue by ozett** [uru-card/uru-card#10]
i would like to build an ESP32-Fido stick with your firmware.
Is your firmware Bluetooth only?
Or will the esp32 be recognized as a USB-Hid device after flashing, means as a USB-Fido2 stick?

thanks, regrads,
toz

by the way: all your projects look interesting, i will dig a bit...

**Comment by snakeye** [comment]
At the moment it's BLE only and the main issue here is the lack of a USB controller in the ESP32 MCU. 

I'm looking for the possibility to implement USB HID using other microcontrollers. Unfortunately, there is not much progress so far.

**Comment by ozett** [comment]
thanks for clarification...

looks like HID is not coming tomorrow..
https://esp32.com/viewtopic.php?t=19949


edit: or tommorow?
https://github.com/chegewara/EspTinyUSB/blob/master/examples/device/hid/keyboard/keyboard.ino

**Comment by snakeye** [comment]
Oh, the second one looks very promising! I did not see it yet. Will give it a try as well!

**Comment by snakeye** [comment]
The second library is designed for ESP32-S2 chips. It has a USB controller on board but does not have BLE support. That means a completely different hardware version for the USB stick. I've ordered a development board with ESP32-S2 to see the library in action.

**Comment by ozett** [comment]
Hey, cool...
i will stay tuned ...

**Comment by ozett** [comment]
seems that the whole s2 family has usb-OTG support... 
https://www.espressif.com/sites/default/files/documentation/esp32-s2_datasheet_en.pdf

**Comment by ozett** [comment]
i discovered that S3 Models seems have two USB Ports (one for HID) ... 
https://www.adafruit.com/?q=s3&sort=BestMatch

**Comment by snakeye** [comment]
One USB Port is more than enough

**Comment by ozett** [comment]
> One USB Port is more than enough

yes, but i saw a adafruit-video and had the impression that one port is only for flashing/config 
and the other port is the HID port. interesting to see, if it could provide some fido-key-functionality

i also orderd a esp-s2, which has only 1 port soldered on, and will see what HID functions i will bring

**Comment by snakeye** [comment]
I've received ESP32-S2 recently. Unfortunately, the current code + PlatformIO does not compile and need some refactoring.

**Comment by ozett** [comment]
nice little S2 board, 

https://www.adafruit.com/product/5325

coming to shop soon, the guys said on youtube.

(annotation mentioning the usb-support)

**Comment by ozett** [comment]
china has delivered the S2 (with Usb-C, look like)

![image](https://user-images.githubusercontent.com/5937811/158451132-332ebfef-c660-4711-8ba1-f28e376bb047.png)

**Comment by ozett** [comment]
[first step](https://www.wemos.cc/en/latest/tutorials/s2/get_started_with_micropython_s2.html#) with the board...


![image](https://user-images.githubusercontent.com/5937811/158576297-25b1f792-58f1-444f-adc1-9bb62bf2eb70.png)

**Comment by ozett** [comment]
this could be an example for an usecase for usb-hid and fingerprint
but with little bit more security ..

https://github.com/MichaelDworkin/Fingerprint-Macro-Key-USB-HID

**Comment by snakeye** [comment]
Yes, I have experimented with UART fingerprint sensors. It's very easy to attach them to the microcontroller but the security is questionable.
